services/provider/consumerstatus/status.go

```diff
@@ -0,0 +1,41 @@
+package consumerstatus
+
+import (
+	pb "github.com/red-hat-storage/ocs-operator/v4/services/provider/pb"
+)
+
+type ConsumerStatus interface {
+	GetClusterVersion() string
+	GetOperatorVersion() string
+
+	SetClusterVersion(string) ConsumerStatus
+	SetOperatorVersion(string) ConsumerStatus
+}
+
+type StatusInfo struct {
```

Comment:
1. This should be a private type (starts with a lowercase), not a public one.
2. The name is not descriptive enough and does not fit the usage
3. The implementation needs to go into this package `github.com/red-hat-storage/ocs-operator/v4/services/provider/pb` and not into its own package to allow that package to access the internals of the struct.
4. I am not sure we need a complete package just for an interface (unless we need to avoid cyclic references) 


```suggestion
type consumerStatusProtobufWrapper struct {
```

Reply:
4. we need interface in separate pkg as it's being imported in multiple pkgs internally and outside of this module as well

worked on rest of the points.